### PR TITLE
fetch configs in reporter

### PR DESCRIPTION
**Pull request by Intizar-T**
# Description

fetch configs instead of querying from db

Related issue #1777 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: node/pkg/reporter/main_test.go [hunk 1]
```diff
@@ -13,50 +13,45 @@ import (
...
 	var res []int32
 	for _, config := range configs {
-		insertedConfig, err := db.QueryRow[Config](ctx, InsertConfigQuery, map[string]any{"name": config.Name, "aggregate_interval": config.AggregateInterval, "submit_interval": config.SubmitInterval})
+		insertedConfig, err := db.QueryRow[int32](ctx, InsertConfigQuery, map[string]any{"name": config.Name, "submit_interval": config.SubmitInterval})
 		if err != nil {
 			log.Error().Err(err).Msgf("error inserting config %s", config.Name)
 			return nil, err
 		}
-		res = append(res, insertedConfig.ID)
+		res = append(res, insertedConfig)
```
[the remark this answers is not in the document]
Comment (Intizar-T) [reply node/pkg/reporter/main_test.go]: nope you are not missing anything, i guess i'm being sleepy hahah. Removed db from tests as well 😄

File: node/taskfiles/taskfile.local.yml [hunk 2]
```diff
@@ -105,7 +105,7 @@ tasks:
   test-reporter:
     dotenv: [".env"]
     cmds:
-      - go test ./pkg/reporter -v
+      - go test ./pkg/reporter -v -count=1
```
Comment (nick-bisonai) [inline node/taskfiles/taskfile.local.yml:108]: is -count=1 necessary here? 😅

Comment (Intizar-T) [reply node/taskfiles/taskfile.local.yml:108]: ahh i keep forgetting it since I need it locally 😢

**Comment by nick-bisonai** [comment]
@coderabbitai review

File: node/pkg/reporter/app.go [hunk 3]
```diff
@@ -124,13 +125,19 @@ func (a *App) startReporters(ctx context.Context) {
 	}
 }
 
-func getConfigs(ctx context.Context) ([]Config, error) {
-	reporterConfigs, err := db.QueryRows[Config](ctx, GET_REPORTER_CONFIGS, nil)
+func fetchConfigs() ([]Config, error) {
+	chain := os.Getenv("CHAIN")
+	if chain == "" {
+		log.Info().Str("Player", "Reporter").Msg("CHAIN env not set, defaulting to baobab")
```
Comment (nick-bisonai) [inline node/pkg/reporter/app.go:131]: let's remove error log here, it'll probably be printed again from line 66,

File: node/pkg/reporter/app.go [hunk 4]
```diff
@@ -124,13 +125,19 @@ func (a *App) startReporters(ctx context.Context) {
...
+		chain = "baobab"
+	}
+	endpoint := fmt.Sprintf("https://config.orakl.network/%s_configs.json", chain)
+	configs, err := request.Request[[]Config](request.WithEndpoint(endpoint))
 	if err != nil {
-		log.Error().Str("Player", "Reporter").Err(err).Msg("failed to load reporter configs")
+		log.Error().Str("Player", "Reporter").Err(err).Msgf("failed to fetch configs from endpoint %s", endpoint)
```
Comment (nick-bisonai) [inline node/pkg/reporter/app.go:137]: let's remove error log here, it'll probably be duplicate logging from line 66

Comment (Intizar-T) [reply node/pkg/reporter/app.go]: done!

File: node/pkg/reporter/app_test.go [hunk 5]
```diff
@@ -11,110 +11,75 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestFetchConfigs(t *testing.T) {
+	configs, err := fetchConfigs()
+	if err != nil || configs == nil || len(configs) == 0 {
+		t.Fatalf("error getting configs: %v", err)
+	}
```
Comment (nick-bisonai) [inline node/pkg/reporter/app_test.go:18]: it's not a big deal, just an optional comment
you can probably use assert.Nil here

something like
```
assert.Nil(t, err)
```

**Review by nick-bisonai — APPROVED** [review]
lgtm!

**Comment by nick-bisonai** [comment]
code rabiit is taking some time lol

File: node/pkg/reporter/app_test.go (hunk 5 above)
Comment (Intizar-T) [reply node/pkg/reporter/app_test.go:18]: if you don't insist, i think i'll keep it like this for now since it's a nice one liner :)

Comment (nick-bisonai) [reply node/pkg/reporter/app_test.go:18]: sure

Comment (nick-bisonai) [reply node/pkg/reporter/app_test.go:18]: but in this case we can't tell between by which of 3 cases it failed

**Comment by Intizar-T** [comment]
> code rabit is taking some time lol

i think it already approved earlier... should I go ahead and merge or wait until it reviews?

**Comment by nick-bisonai** [comment]
> > code rabit is taking some time lol
> 
> i think it already approved earlier... should I go ahead and merge or wait until it reviews?

that shouldn't matter but yh, go ahead

**Pull request merged**